src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -824,7 +824,10 @@ public string ConstructPrimaryKeyRoute(RestRequestContext context, Dictionary<st
                     parameters = new(context.PrimaryKeyValuePairs!);
                     foreach (KeyValuePair<string, object?> pair in context.FieldValuePairsInBody)
                     {
-                        parameters.Add(pair.Key, pair.Value);
+                        // Use TryAdd because there can be primary key fields present in the request body as well
+                        // (in addition to request URL), when we operate in non-strict mode for REST.
+                        // In such a case, the duplicate PK fields in the request body are ignored.
+                        parameters.TryAdd(pair.Key, pair.Value);
```

Comment:
This static function just populates all the parameters into the params dictionary. Correct, this would added publisherName as well here. The publisherName would however be filtered out when we create the actual SqlUpdate/Insert/Upsert etc. structure. You can refer to the changes made to those files.

Reply:
Correct, publisherName would be added here. However it will get filtered out later when we build the respective Sql{*}QueryStructure for the mutation. However, since we support extraneous fields for REST only, we can do filtering here itself. I will make the necessary change.

Reply:
Done